### ISSUE TITLE
Authentication broken on latest OCM

### ISSUE THREAD

**Issue by Skarlso** [open-component-model/ocm#368]
**What happened**:

```
ocm get resources ghcr.io/skarlso/mpas//mpas.ocm.software/podinfo/backend -c 1.0.0
NAME         VERSION IDENTITY TYPE                        RELATION
config       1.0.0            configdata.ocm.software     local
image        6.2.0            ociImage                    external
instructions 1.0.0            PlainText                   local
manifests    1.0.0            kustomize.ocm.fluxcd.io     local
validation   1.0.0            validator.mpas.ocm.software local
➜  mpas-product-controller-examples ocm version
version.Info{Major:"0", Minor:"3", Patch:"0", PreRelease:"", Meta:"48b6b2999515bc45f96cac602d35e99b34374c27", GitVersion:"v0.3.0+48b6b2999515bc45f96cac602d35e99b34374c27", GitCommit:"48b6b2999515bc45f96cac602d35e99b34374c27", GitTreeState:"clean", BuildDate:"2023-05-15T17:02:55+0000", GoVersion:"go1.20.4", Compiler:"gc", Platform:"darwin/amd64"}
➜  mpas-product-controller-examples ocm get resources ghcr.io/skarlso/mpas//mpas.ocm.software/podinfo/backend -c 1.0.0
Error: error processing "ghcr.io/skarlso/mpas//mpas.ocm.software/podinfo/backend": failed to authorize: failed to fetch oauth token: unexpected status: 405 Method Not Allowed
➜  mpas-product-controller-examples ocm version
{"Major":"0", "Minor":"3", "Patch":"0", "PreRelease":"dev", "Meta":"8c3f49de694c06b340ca26e72e8fe7841a9b8ef0", "GitVersion":"0.3.0-dev+8c3f49de694c06b340ca26e72e8fe7841a9b8ef0", "GitCommit":"8c3f49de694c06b340ca26e72e8fe7841a9b8ef0", "GitTreeState":"clean", "BuildDate":"2023-05-16T07:07:50+0000", "GoVersion":"go1.20.4", "Compiler":"gc", "Platform":"darwin/amd64"}
➜  mpas-product-controller-examples
```

Config exists `./ocmconfig`:

```
type: generic.config.ocm.software/v1
configurations:
- type: credentials.config.ocm.software
  repositories:
    - repository:
        type: DockerConfig/v1
        dockerConfigFile: "~/.docker/config.json"
        propagateConsumerIdentity: true
- type: attributes.config.ocm.software
  attributes:
    cache: ~/.ocm/cache
```

**What you expected to happen**:

Authentication to work.

**How to reproduce it (as minimally and precisely as possible)**:

Use ocm v0.3.0 first, then use latest `main` with the above commands.

**Anything else we need to know**:

**Environment**:

**Comment by Skarlso** [comment]
Unsetting `GITHUB_TOKEN` environment property made it work. But since I will always have a GITHUB_TOKEN in my env, and it is still working in v0.3.0 it's not a preferred solution. :)

**Comment by morri-son** [comment]
fixed with #372
